### PR TITLE
Do not set CrossBuild to true when building runtime tests for Android

### LINKED ISSUE #55669: _packageRID has wrong value for runtime tests build for Android Arm64 and Arm32
When building runtime tests for Android arm64 or arm32, the value assigned to `_packageRID` has become `linux-arm64` lately. It should be `android-arm64`. This has been causing CI lane `Build Android arm64 Release AllSubsets_Mono_RuntimeTests` failing with error message
```
/__w/1/s/eng/liveBuilds.targets(134,5): error : The 'libs' subset must be built before building this project. Missing artifacts: /__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-arm64/Release/runtimes/linux-arm64/lib/net6.0/. Configuration: 'Release'. To use a different configuration, specify the 'LibrariesConfiguration' property.
```

Here is the link to one of the failing build: https://dev.azure.com/dnceng/public/_build/results?buildId=1237622&view=logs&j=1fbc174f-133f-57a8-dcec-0de15ad033b4&t=827c0caa-5f6d-5c33-e064-1c275c207b92&l=51

This might be caused by a recent change to the following line of code
https://github.com/dotnet/runtime/blob/main/Directory.Build.props#L166

The related PR is https://github.com/dotnet/runtime/pull/54914

### PR DESCRIPTION
**Pull request by fanyang-mono**
Fixes #55669

### CODE REVIEW

**Comment by fanyang-mono** [comment]
This change makes FreeBSD build failing though. Need to investigate further.

**Comment by ViktorHofer** [comment]
cc @am11

**Comment by am11** [comment]
Strange that it didn't showed up in CI runs. Was it excluded for some other reason?
PackageRID is the RID to restore package. OutputRid is the RID to create shipping package.

If changing package RID is fixing the "Building Targeting Pack" issue on Android, while all other platforms are fine, I think we should find the reason why Android with CrossBuild=true is expecting package restore RID to be same as target, rather than host?

**Comment by am11** [comment]
Does deleting this override line fix Android test build: https://github.com/dotnet/runtime/blob/b937677e8f8601848d29bc072a93cc0c6e21576d/src/tests/run.proj#L651?

**Comment by fanyang-mono** [comment]
> Does deleting this override line fix Android test build:
> 
> https://github.com/dotnet/runtime/blob/b937677e8f8601848d29bc072a93cc0c6e21576d/src/tests/run.proj#L651
> 
> ?

No the failure happened before reaching to this point.

**Comment by fanyang-mono** [comment]
> Strange that it didn't showed up in CI runs. Was it excluded for some other reason?
> PackageRID is the RID to restore package. OutputRid is the RID to create shipping package.
> 
> If changing package RID is fixing the "Building Targeting Pack" issue on Android, while all other platforms are fine, I think we should find the reason why Android with CrossBuild=true is expecting package restore RID to be same as target, rather than host?

The Android arm64 lane only run on rolling build. That's why it didn't fail for the PR checks.

**Comment by am11** [comment]
> The Android arm64 lane only run on rolling build. That's why it didn't fail for the PR checks.

Is rolling build different than the official build? We did ran official build in that PR.

> No

Alright, then this particular leg of Android is different than anything else we have in PR, official and unofficial platform matrix. We are not passing -cross and setting /CrossBuild=false, which it is clearly cross-compilation on Ubuntu for Android. I think we should try to fix this leg's configuration, rather than changing the meaning of things again.

**Comment by fanyang-mono** [comment]
> Is rolling build different than the official build? We did ran official build in that PR.
Rolling build is the build against main, which ran a couple times per day. It is not triggered as often as for every PR.

> Alright, then this particular leg of Android is different than anything else we have in PR, official and unofficial platform matrix. We are not passing -cross and explicitly setting /CrossBuild=false, when it is clearly cross-compilation on Ubuntu for Android. I think we should try to fix this leg's configuration, rather than changing the meaning of things again.

I just did some more investigation and found that for the android arm64 runtime test build, both `CrossBuild` and `PortableBuild` were set to true. I updated the condition for `_packageRID` to accommodate this scenario.

**Comment by ViktorHofer** [comment]
> Is rolling build different than the official build? We did ran official build in that PR.

Unfortunately yes but fortunately they are public builds. Rolling builds are ones that are triggered when a PR is merged, they run on the main branch. Because of the low number of hardware available for certain configurations we can't run all configurations per PR which is why some only run on rolling builds. We usually only have about 10-15 rolling builds per day (as those are batched together as well).

> I just did some more investigation and found that for the android arm64 runtime test build, both CrossBuild and PortableBuild were set to true

Why is PortableBuild=true passed in at all? It's already true by default: https://github.com/dotnet/runtime/blob/e60882e13769df03efe6c564311b9d3793e693d6/Directory.Build.props#L111

**Comment by fanyang-mono** [comment]
> Why is PortableBuild=true passed in at all?
It was set at
https://github.com/dotnet/runtime/blob/main/src/tests/build.sh#L362
and
https://github.com/dotnet/runtime/blob/main/src/tests/build.sh#L400

The runtime test infrastructure is a complicated beast. Actually, I don't know exactly why it `PortableBuild=true` needs to be passed in explicitly. Maybe we don't need to. Even thought it was not passed in explicitly, `PortableBuild` will still be true. And the problem that this PR is trying to solve will still be there.

**Comment by am11** [comment]
> I just did some more investigation and found that for the android arm64 runtime test build, both `CrossBuild` and `PortableBuild` were set to true. I updated the condition for `_packageRID` to accommodate this scenario.

Ah, ok. I was looking at (https://dev.azure.com/dnceng/public/_build/results?buildId=1237622&view=logs&j=1fbc174f-133f-57a8-dcec-0de15ad033b4&t=827c0caa-5f6d-5c33-e064-1c275c207b92) Build Product step and there we have:

` /__w/1/s/build.sh -ci -arch arm64 -os Android -s mono+libs -c Release  /p:CrossBuild=false`

I don't know how to explain it based on what we understood. If we are cross compiling then /p:CrossBuild should be true (or just pass -cross) and ROOTFS_DIR environment variable should be set. Similarly, in test build step:

`/__w/1/s/src/tests/build.sh /p:LibrariesConfiguration=Release -ci -excludemonofailures os Android arm64 /p:RuntimeVariant= Release`

should have -cross flag set rather than setting it somewhere in the middle of the build? Then we won't be needing one-off workarounds like https://github.com/dotnet/runtime/blob/b937677e8f8601848d29bc072a93cc0c6e21576d/src/tests/run.proj#L651

if something is not working with ROOTFS_DIR and -cross (which has been the cross build way), we can investigate it and fix it to make it work.

File: src/tests/build.sh [hunk 1]
```diff
@@ -573,7 +573,7 @@ if [[ "${__BuildArch}" != "${__HostArch}" ]]; then
     __CrossBuild=1
 fi
 
-if [[ "$__CrossBuild" == 1 ]]; then
+if [[ ("$__CrossBuild" == 1) && ("$__TargetOS" != "Android") ]]; then
```
Comment (am11) [inline src/tests/build.sh:576]: nit:
```suggestion
if [[ "$__CrossBuild" == 1 && "$__TargetOS" != "Android" ]]; then
```

**Review by am11 — COMMENTED** [review]
Thank you! Looks great. :)

Android cross-build seems a bit different than other platforms, as while it is a cross build, we don't set -cross arguments (and set `ROOTFS_DIR` environment variable) for it. It is probably to workaround something, but IMHO, we should try to fix the reason of workaround at some point, so we could cross-build mono as well as coreclr for Android alike (#32800).

**Comment by fanyang-mono** [comment]
> Thank you! Looks great. :)
> 
> Android cross-build seems a bit different than other platforms, as while it is a cross build, we don't set -cross arguments (and set `ROOTFS_DIR` environment variable) for it. It is probably to workaround something, but IMHO, we should try to fix the reason of workaround at some point, so we could cross-build mono as well as coreclr for Android alike (#32800).

Do you mean the build for Mono Android product itself or the runtime test build script that I am fixing here?

**Comment by am11** [comment]
> Do you mean the build for Mono Android product itself or the runtime test build script that I am fixing here?

You have added an exception in test build script which reads "do not consider Android's cross build as *typical* cross-build".
So whatever workaround'ish reason it entails is what I was referring to. `-cross` is neither passed to Android's product build nor test build, while it's clearly a cross-build.

**Comment by fanyang-mono** [comment]
> > Do you mean the build for Mono Android product itself or the runtime test build script that I am fixing here?
> 
> You have added an exception in test build script which reads "do not consider Android's cross build as _typical_ cross-build".
> So whatever workaround'ish reason it entails is what I was referring to. `-cross` is neither passed to Android's product build nor test build, while it's clearly a cross-build.

Ah, gotcha!

**Comment by fanyang-mono** [comment]
@am11 If this change looks good to you, do you mind approving it?

**Review by am11 — APPROVED** [review]

**Comment by am11** [comment]
@ViktorHofer if you could give a green light on this too that would be great. :)

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**